tests: safer comparison

See Travis CI build failure:

```
tests/test_ext.py:152: in run
    '{0}'.format(expected_data))
E   AssertionError: "{'baa': None, 'bar': 'foo'}" != "{'bar': 'foo', 'baa': None}"
E   - {'baa': None, 'bar': 'foo'}
E   + {'bar': 'foo', 'baa': None}
```
https://travis-ci.org/inveniosoftware/flask-sso/jobs/161349044

--- a/tests/test_ext.py
+++ b/tests/test_ext.py
@@ -9,6 +9,7 @@
 
 from __future__ import absolute_import
 
+from ast import literal_eval
 from contextlib import contextmanager
 
 from flask import request, request_started
@@ -76,8 +77,8 @@ class TestSSO(FlaskTestCase):
             with request_environ_set(self.app, data):
                 with self.app.test_client() as c:
                     resp = c.get(self.app.config['SSO_LOGIN_URL'])
-                    self.assertEqual(resp.data.decode('utf-8'),
-                                     '{0}'.format(expected_data))
+                    assert literal_eval(resp.data.decode('utf-8')) \
+                        == expected_data
 
         conf = {'FOO': (True, 'bar'), 'BAZ': (False, 'baa')}
         data = {'FOO': 'foo'}

